lms/djangoapps/course_api/tests/test_views.py

```diff
@@ -100,6 +103,16 @@ def test_not_logged_in(self):
         self.client.logout()
         self.verify_response()
 
+    @ddt.data('staff', 'user')
+    def test_throttle_is_set_correctly(self, user_scope):
```

Comment:
@robrap I added this test instead of the other one which was in previous PR but it's not actually testing the API endpoint is being throttled or not. Is that okay?

Reply:
The issue with the other test is that it will become flaky over time if the rate limit ever gets increased. In the test you created for the last PR, it wasn't as much as an issue because we were dealing with 2 and 10/minute for the rate limit. In the past, when the rate limit was incrementally increased to 2400, the test started to time out and was eventually deemed flaky.

Reply:
@waheedahmed: This looks good.

If we decide we want more of an integration test to ensure that throttling actually occurs (which I'm not sure whether we need), another option would be to have the test force the rate to be 0 (or 1) and ensures that the loop is called twice, once passing and once with a throttle error, rather than basing the test on the the hard coded throttle rate.  I don't think we need this, but some future RCA could always prove me wrong.

The current test assumes the switches are False.  Two more temporary tests that duplicate this test, but turn a waffle switch on, would ensure all combos are configured properly.  NOTE: We had 2 recent CAT-1s because of config issues.

Additionally, we could have something like the following untested code:
```
@ddt.data(('staff', False), ('user', False), ('unknown', True))
@ddt.unpack
    def test_throttle_is_set_correctly(self, user_scope, throws_exception):
        ...
```
This would ensure that this class is a proper throttle class and it is raising the ImproperlyConfigured exception as well.